### PR TITLE
Port nullability annotations to refs XmlDocument and XmlSerializer

### PR DESCRIPTION
**Pull request by jozkee**
Contributes to https://github.com/dotnet/runtime/issues/2339

For System.Xml.XmlDocument all types were forwarded to ReaderWriter assembly so I only enabled nullability in the csprojs.
For System.Xml.XmlSerializer some types were forwarded to ReaderWriter but some other types were in place.

### CODE REVIEW

**Review by jeffhandley — APPROVED** [review]
LGTM but please get a second review

File: src/libraries/System.Data.Common/src/System/Data/Common/ObjectStorage.cs
Comment (krwq) [inline src/libraries/System.Data.Common/src/System/Data/Common/ObjectStorage.cs:344]: wondering if this should return nullable instead but perhaps might be better to file an issue rather than solve here

File: src/libraries/System.Data.Common/src/System/Data/Common/SqlUDTStorage.cs
Comment (krwq) [inline src/libraries/System.Data.Common/src/System/Data/Common/SqlUDTStorage.cs:164]: also IMO we should file an issue to investigate if this should return nullable here and below

File: src/libraries/System.Xml.XmlSerializer/ref/System.Xml.XmlSerializer.cs
Comment (krwq) [inline src/libraries/System.Xml.XmlSerializer/ref/System.Xml.XmlSerializer.cs:99]: wondering if this should be nullable arg considering property is allowing null

**Review by krwq — APPROVED** [review]

File: src/libraries/System.Xml.XmlSerializer/ref/System.Xml.XmlSerializer.cs
Comment (Jozkee) [reply src/libraries/System.Xml.XmlSerializer/ref/System.Xml.XmlSerializer.cs:99]: Agree.

File: src/libraries/System.Data.Common/src/System/Data/Common/SqlUDTStorage.cs
Comment (Jozkee) [reply src/libraries/System.Data.Common/src/System/Data/Common/SqlUDTStorage.cs:164]: Created https://github.com/dotnet/runtime/issues/41497.

**Comment by krwq** [comment]
LGTM, please file an issue for the potential System.Data.Common issue (or TODO-NULLABLE)

**Pull request merged**
